### ISSUE TITLE
[CoE Starter Kit - QUESTION] August 2024 Release - Why are we using AuditLogQuery.Read.All Permission?

### ISSUE THREAD

**Issue by KeysAU** [microsoft/coe-starter-kit#8873]
### Does this question already exist in our backlog?

- [X] I have checked and confirm this is a new question.

### What is your question?

For the new august release Why are we using AuditLogQuery.Read.All with the new graph API permissions instead of specifying the specific logs the COE kit needs?

It seems unnecessary for the COE kit to have access to everyone's Exchange or OneDrive logs. Wouldn't it be more secure to follow the principle of least privilege and only request access to the required audit logs?

Here's more information: [Graph API Audit Log Queries Permissions](https://learn.microsoft.com/en-us/graph/api/security-auditcoreroot-post-auditlogqueries?view=graph-rest-beta&tabs=http&WT.mc_id=M365-MVP-9501#permissions)

<img width="516" alt="image" src="https://github.com/user-attachments/assets/e2461d63-b727-4aad-93a0-1baaa0408f60">
 

### What solution are you experiencing the issue with?

Audit Log

### What method are you using to get inventory and telemetry?

None

**Comment by Jenefer-Monroe** [comment]
Thanks for posting, we'll take a look to see if that works.

**Comment by pete-msft** [comment]
@KeysAU Good question! Goal is to always use the least privileged permissions but currently there isn't dedicated permission for Power Platform workload, so we need to use AuditLogsQuery.Read.All

https://learn.microsoft.com/en-us/graph/permissions-reference

**Comment by KeysAU** [comment]
> @KeysAU Good question! Goal is to always use the least privileged permissions but currently there isn't dedicated permission for Power Platform workload, so we need to use AuditLogsQuery.Read.All
> 
> https://learn.microsoft.com/en-us/graph/permissions-reference

Thanks for the response, Pete.

Can you please flag this as a security concern internally with the responsible Microsoft product team? The current 'least permission' setting is equivalent to 'Global Audit Log Reader,' as there isn't a dedicated permission set for Power Platform logs. 

Practically speaking, this is not a true 'least privilege' setting.

**Comment by pete-msft** [comment]
@KeysAU Yes we have discussions ongoing currently with PG about this. And yes you are right, it is not true least privilege configuration but like said that is the permission we unfortunately need to use now. This is the same level of permissions when using Office 365 Management API option to gather the logs. In that legacy API we only have similar read all events permission and CoE just queries general events which does not contain EXO, SPO and Entra

![image](https://github.com/user-attachments/assets/7afb7bf1-e3e9-44a8-8eab-29515a2b32fc)


![image](https://github.com/user-attachments/assets/f0c7deb3-a877-40cd-8209-5fb8b7b36c39)

**Comment by Jenefer-Monroe** [comment]
Let's use this issue to track by assigning to Grant (who owns pushing these to PG) and marking as product gap

**Comment by perssonkasia** [comment]
Helle, we're also encountering the same concern with the AuditLogQuery.Read.All permission. Does Microsoft have any plans to introduce a more scoped permission, for example, limiting AuditLogQuery.Read.All to just Power Platform audit logs? Also, is there any known workaround in the meantime?
Appreciate any insights or guidance you can share 🙏

**Comment by IvanFAA** [comment]
[like]  Amir, Ivan (FAA) reacted to your message:
________________________________

**Comment by baur2** [comment]
Hello,
we are also facing the same issue. Are there any new developments on this topic?
